server/db/src/main/java/com/walmartlabs/concord/db/migration/SecretsHashMigrationTask.java

```diff
@@ -0,0 +1,105 @@
+package com.walmartlabs.concord.db.migration;
+
+/*-
+ * *****
+ * Concord
+ * -----
+ * Copyright (C) 2017 - 2022 Walmart Inc.
+ * -----
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =====
+ */
+
+import com.walmartlabs.concord.common.secret.HashAlgorithm;
+import com.walmartlabs.concord.common.secret.SecretEncryptedByType;
+import com.walmartlabs.concord.common.secret.SecretUtils;
+import com.walmartlabs.ollie.config.Config;
+import org.postgresql.util.PGobject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.UUID;
+
+@Named
+@MigrationOrder(1)
+public class SecretsHashMigrationTask implements MigrationTask {
+
+
+    private static final Logger log = LoggerFactory.getLogger(SecretsHashMigrationTask.class);
+    private static final int MAX_RETRIES = 10;
+    private static final int BATCH_SIZE = 100;
+
+    @Inject
+    @Config("secretStore.serverPassword")
+    private byte[] serverPwd;
+
+    @Override
+    synchronized public void execute(DataSource dataSource) {
+        log.info("Starting migration task for secretsHash");
+        int retryCount = 0;
+        while(retryCount <= MAX_RETRIES) {
+            try(Connection con = dataSource.getConnection()) {
+                while(true) {
+                    try(PreparedStatement psSelect = con.prepareStatement("SELECT secret_id, secret_salt, secret_data FROM secrets WHERE hash_algorithm = ? AND encrypted_by = ? LIMIT ?")){
```

Comment:
I think there should be a "select to update" so that no one can change the value during the update.

Reply:
Updated